src/encoded/schemas/quality_metric.json

```diff
@@ -55,7 +56,8 @@
                     "title": "MIME type",
                     "type": "string",
                     "enum": [
-                        "text/plain"
+                        "text/plain",
+                        "image/png"
```

Comment:
I see.  Yes explicitly stating the type in the schema (rather than getting it from the mixin) is better.  The way I have it with the enum in the mixin you could end up with a particular QC object type that relies on the mixin for the mime type, and get some instances of the object with “attachment” properties as text and some as image.  That’s probably not good because consumers of that object type wouldn’t necessarily know what mime type to expect, just looking at the schema.

In fact, for ChIP, all but one of the QC objects have image attachments.  But those are (or will be soon) named something other than “attachment” anyway.

So, yes, keeping this just text is fine and I now see why you did it this way in the first place.

On Feb 2, 2016, at 5:01 PM, Laurence Rowe <notifications@github.com<mailto:notifications@github.com>> wrote:


In src/encoded/schemas/quality_metric.json<https://github.com/ENCODE-DCC/encoded/pull/1136#discussion_r51663992>:

> @@ -55,7 +56,8 @@
>                      "title": "MIME type",
>                      "type": "string",
>                      "enum": [
> -                        "text/plain"
> +                        "text/plain",
> +                        "image/png"


Mixins are just a convenience for constructing the final schemas. If there are only a few quality metrics that support image, we should leave the default be text and override the attachment property only on those. If it's common then making it generic like this is fine. Being overly restrictive about the attachment type probably doesn't buy us much.

—
Reply to this email directly or view it on GitHub<https://github.com/ENCODE-DCC/encoded/pull/1136/files#r51663992>.